central/processlisteningonport/datastore/datastore_impl.go

```diff
@@ -295,14 +295,24 @@ func (ds *datastoreImpl) fetchIndicators(
 			continue
 		}
 
-		indicatorLookups = append(indicatorLookups,
-			search.NewQueryBuilder().
-				AddExactMatches(search.ContainerName, val.Process.ContainerName).
-				AddExactMatches(search.PodID, val.Process.PodId).
-				AddExactMatches(search.ProcessName, val.Process.ProcessName).
-				AddExactMatches(search.ProcessArguments, val.Process.ProcessArgs).
-				AddExactMatches(search.ProcessExecPath, val.Process.ProcessExecFilePath).
-				ProtoQuery())
+		if val.Process.ProcessArgs != "" {
+			indicatorLookups = append(indicatorLookups,
+				search.NewQueryBuilder().
+					AddExactMatches(search.ContainerName, val.Process.ContainerName).
+					AddExactMatches(search.PodID, val.Process.PodId).
+					AddExactMatches(search.ProcessName, val.Process.ProcessName).
+					AddExactMatches(search.ProcessArguments, val.Process.ProcessArgs).
+					AddExactMatches(search.ProcessExecPath, val.Process.ProcessExecFilePath).
+					ProtoQuery())
+		} else {
+			indicatorLookups = append(indicatorLookups,
+				search.NewQueryBuilder().
+					AddExactMatches(search.ContainerName, val.Process.ContainerName).
+					AddExactMatches(search.PodID, val.Process.PodId).
+					AddExactMatches(search.ProcessName, val.Process.ProcessName).
+					AddExactMatches(search.ProcessExecPath, val.Process.ProcessExecFilePath).
+					ProtoQuery())
+		}
```

Comment:
There is probably a better way of doing this. I tried using AddNullField in the else block, but that didn't work. It should at least be possible to factor out the common code.